### ISSUE TITLE
Build: show the command that's currently being executed

### ISSUE THREAD

**Issue by stsewd** [readthedocs/readthedocs.org#4288]
Currently, we save the command after running it, which has this two problems:

- The user can't see the current command that is running (their only see the previous command!)
- If the command it's killed due timeout limits (or another uncaught exception), the command that originates the problem isn't recorded. This lead to a false clue of where the problem is.

I have played a little with the save command and the API, is possible to do this, we only need to be careful with the data that is saved before the command is executed (for the current API some values like `end_time` can't be None).

ref https://github.com/rtfd/readthedocs.org/issues/3884 and other related issues

**Comment by humitos** [comment]
I think it makes sense to save the command _before_ running with just the command that's going to be executed and _after_ it's has finished with all the details (end time, exit code, etc) or adding a custom message in the output if the command has failed. I'm not sure how easy would be the latter.

**Comment by humitos** [comment]
This issue is related (maybe duplicated) to an old one: #1863

**Comment by humitos** [comment]
This issue is related (maybe duplicated) to an old one: #1863

**Comment by ericholscher** [comment]
It seems like this is going to 2x the amount of API calls we'll need for every build, with each command now requiring a "start" and "end" API call?

**Comment by stsewd** [comment]
Yeah, but we are using a patch method, so less info. And we already call the api a lot to do the _live report_ (but that's a read only operation)

**Comment by humitos** [comment]
> And we already call the api a lot to do the _live report_ (but that's a read only operation)

But the _live report_ is _only for users_ **watching** the build.

**Comment by ericholscher** [comment]
I had a user request this, and think it's definitely something worth doing, especially for long-running build steps like the HTML build. It is kinda weird that it just spins for ~20 minutes for some longer builds without showing what is happening.
